src/tree/containers/ContainersTreeItem.ts

```diff
@@ -122,6 +122,19 @@ export class ContainersTreeItem extends LocalRootTreeItemBase<DockerContainerInf
         return super.getTreeItemForEmptyList()
     }
 
+    protected areArraysEqual(array1: DockerContainerInfo[] | undefined, array2: DockerContainerInfo[] | undefined): boolean {
+        if (!super.areArraysEqual(array1, array2)) {
+            // If they aren't the same according to the base class implementation, they are definitely not the same according to this either
+            return false;
+        }
+
+        // Containers' labels/descriptions (status in particular) can change. If they do, we want to cause a refresh. But, we also don't want to change the tree ID based on status.
```

Comment:
The part about tree ID: IMO it is a bit unclear how it relates to the code in this file. Suggest mentioning, or adding a reference to `getTreeId()` function.